src/tests/unit/tests/electron/views/automated-checks/automated-checks-view.test.tsx

```diff
@@ -1,25 +1,47 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { ScanActionCreator } from 'electron/flux/action-creator/scan-action-creator';
+import { ScanStatus } from 'electron/flux/types/scan-status';
 import { AutomatedChecksView, AutomatedChecksViewProps } from 'electron/views/automated-checks/automated-checks-view';
 import { shallow } from 'enzyme';
 import * as React from 'react';
 import { Mock, Times } from 'typemoq';
 
 describe('AutomatedChecksView', () => {
-    it('renders the automated checks view', () => {
-        const props: AutomatedChecksViewProps = {
-            deps: {
-                deviceConnectActionCreator: null,
-                currentWindow: null,
-                windowStateActionCreator: null,
-                scanActionCreator: Mock.ofType(ScanActionCreator).object,
-            },
-        } as AutomatedChecksViewProps;
+    describe('renders', () => {
+        it('the automated checks view', () => {
+            const props: AutomatedChecksViewProps = {
+                deps: {
```

Comment:
I strongly prefer casting the deps object as the deps necessary in this class.

The reason for this is that as the deps underneath this component changes, you will have to modify this object as well, even if this specific component does not consume that dependency, which minimizes the utility of having deps in the first place. Instead, it's better to only mention the dependencies that this component consumes and then cast that object; this is not something you can miss either as if you consume a dependency and it is not defined in that object (you will miss the type error; no error at compile time), your test will at least still fail at run time.

In this case, the deps would only include scanActionCreator because that is the only consumed dependency in that class.

Let me know if that makes sense.

Reply:
As talked offline: casting at the props level technically speaking also cast the deps level. I did remove the deps we don't need for this unit tests though! Thanks for the feedback.